Expensify/App
# Add Onfido SDKs + Enable Payments flow

### Details
Still Holding on 

**Auth to deploy:**
- [x]  https://github.com/Expensify/Auth/pull/5520

**Web-Secure changes to be merged/deployed:**
- [x] https://github.com/Expensify/Web-Secure/pull/1944

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/161733

### Tests

**Testing tip:** As you work through this it will be useful to reset the wallet data back to the default state. To restart the flow locally we can run this query:

```
sqlite> UPDATE bankAccounts SET additionalData = JSON_SET(COALESCE(NULLIF(additionalData, ''), '{}'), '$.verifications', JSON('{}')) WHERE bankAccountID = 1030799;
sqlite> UPDATE bankAccounts SET additionalData = JSON_SET(COALESCE(NULLIF(additionalData, ''), '{}'), '$.currentStep', 'OnfidoStep') WHERE bankAccountID = 1030799;
```

1. Follow the setup instructions in https://github.com/Expensify/Web-Secure/pull/1944 to get a "pending" wallet
2. Log into E.cash
3. Verify you can access the `/enable-payments` via url or deep link + that the `userWallet` appears in `localStorage` and that you are able to see the Onfido screen after a hopefully brief loading spinner. The wallet itself should have a `SILVER` status.
<img width="1440" alt="2021-05-19_16-53-30" src="https://user-images.githubusercontent.com/32969087/118912099-fc3db200-b8c2-11eb-80ba-16a19b794607.png">

4. Complete the Onfido flow (without using any of your own ID if you can avoid it - Onfido recommends to not use real info) then verify you are brought to the next screen once the flow is completed.
5. Fill out additional details and try to break stuff. Errors should appear on the fields that have issues.
![2021-05-24_09-48-24](https://user-images.githubusercontent.com/32969087/119399970-4208ca80-bc75-11eb-9ba8-318fe6adfb11.png)

6. Before submitting we must mock IDology so it succeeds by altering the line mentioned [here](https://github.com/Expensify/Web-Secure/blob/a23bfb8b57c53b66bb97d43ac0edb971842b5f22/lib/ExpectID/API.php#L74-L75)

7. Next we will should be brought to the Terms placeholder screen (**Note:** Terms will be added in a later step)
![2021-05-24_12-54-53](https://user-images.githubusercontent.com/32969087/119416446-55c12a80-bc8f-11eb-9d1f-4035f80a9bce.png)
8. Try to submit without checking the boxes and verify it's not possible to do so and an error message appears
9. Check the boxes and submit
10. Verify you are brought to a screen that shows the status of your wallet (it should be pending for now since Onfido must be manually approved).
![2021-05-24_10-42-57](https://user-images.githubusercontent.com/32969087/119406183-1e964d80-bc7e-11eb-9689-ec7f51347517.png)

### QA Steps
❌  - This is not yet QA able but probably will have to be production QA and tested in connection with the Web-Secure PR

### Tested On

- Mobile Web - Sort of... Onfido does not seem to be working on mobile web just yet
- Desktop - can't be tested on desktop yet - but should be the same as web 

### Screenshots


#### Web
See test steps

#### Mobile Web
N/A - mobile web Onfido is not quite working yet

#### Desktop
N/A - as it cannot yet be tested

#### iOS
![2021-05-24_11-52-14](https://user-images.githubusercontent.com/32969087/119411956-ef380e80-bc86-11eb-9df7-46afa50974a6.png)
![2021-05-24_11-52-23](https://user-images.githubusercontent.com/32969087/119411966-f101d200-bc86-11eb-8e53-b09b95bb18d3.png)
![2021-05-24_11-52-45](https://user-images.githubusercontent.com/32969087/119411974-f3642c00-bc86-11eb-9afd-edf9ad8dc2f2.png)
![2021-05-24_11-52-55](https://user-images.githubusercontent.com/32969087/119411979-f65f1c80-bc86-11eb-8571-503576189d99.png)
![2021-05-24_11-53-19](https://user-images.githubusercontent.com/32969087/119411987-fb23d080-bc86-11eb-8920-9ae1ccde4909.png)
![2021-05-24_11-53-29](https://user-images.githubusercontent.com/32969087/119411994-fced9400-bc86-11eb-9e60-c764495751b7.png)
![2021-05-24_11-53-38](https://user-images.githubusercontent.com/32969087/119412008-ffe88480-bc86-11eb-9ea6-c69d4c75f766.png)

#### Android
![2021-05-24_12-58-23](https://user-images.githubusercontent.com/32969087/119416639-ccf6be80-bc8f-11eb-9c0e-99a21f3eca2b.png)
![2021-05-24_14-18-01](https://user-images.githubusercontent.com/32969087/119421644-e18c8400-bc9a-11eb-8f72-b8111db92a80.png)
![2021-05-24_12-58-39](https://user-images.githubusercontent.com/32969087/119416647-d2540900-bc8f-11eb-91df-f057e3650493.png)

Test botify
